src/Device/Driver/ATR833.cpp

```diff
@@ -77,14 +123,170 @@ class ATRBuffer {
 };
 
 bool
-ATR833Device::DataReceived(std::span<const std::byte>,
+ATR833Device::DataReceived(std::span<const std::byte> s,
                            [[maybe_unused]] NMEAInfo &info) noexcept
 {
-  // actually made no use of radio information
-  // TODO: interpret data delivered by ATR833 to display Radio settings...
+  assert(!s.empty());
+  const auto *data = s.data();
+  const auto *const end = data + s.size();
+
+  do {
+    // Append new data to the buffer, as much as fits in there
+    auto range = rx_buf.Write();
+    if (rx_buf.IsFull()) {
+      // Overflow: reset buffer to recover quickly
+      rx_buf.Clear();
+      expected_msg_length = 0;
+      continue;
+    }
+    size_t nbytes = std::min(range.size(), size_t(end - data));
+    memcpy(range.data(), data, nbytes);
+    data += nbytes;
+    rx_buf.Append(nbytes);
+
+    for (;;) {
+      // Read data from buffer to handle the messages
+      range = rx_buf.Read();
+      if (range.empty())
+        break;
+
+      if (range.size() < expected_msg_length)
+        break;
+
+      expected_msg_length = ExpectedMsgLength(range.data(), range.size());
+
+      if (range.size() >= expected_msg_length) {
+        if (range.front() == STX) {
+            const std::lock_guard<Mutex> lock(response_mutex);
+            HandleResponse(range.data(), info);
+        }
+        // Message handled -> remove message
+        rx_buf.Consume(expected_msg_length);
+        expected_msg_length = 0;
+        // Received something from the radio -> the connection is alive
+        info.alive.Update(info.clock);
```

Comment:
But who updated `clock`?

Reply:
Ok. Moved the "alive"-update to a more sensible place.